### PR TITLE
fix 'C for Python programmers' link

### PR DESCRIPTION
**Pull request by Spokaren**
## What does this PR do?
Add resource(s)

### Description
The original link is broken, so i replace it by the link from the website of the University of Torento.
## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [ ] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: courses/free-courses-fr.md [hunk 1]
```diff
@@ -23,7 +23,7 @@
 
 ### Algorithmes
 
-* [Cours d'Algorithmie](https://www.youtube.com/playlist?list=PLrSOXFDHBtfGy7xYmf5LlNr8f-niDkf7_) - Formation Video
+* [Algorithmie - Cours](https://www.youtube.com/playlist?list=PLrSOXFDHBtfE0AkOm795c2qpLQJNiEBbZ) - Formation Video
```
Comment (davorpa) [inline courses/free-courses-fr.md:26]: Overscoped? Already merged at #6968

File: books/free-programming-books-langs.md [hunk 2]
```diff
@@ -429,7 +429,7 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 * [Beej's Guide to the GNU Debugger (GDB)](http://beej.us/guide/bggdb/) - Brian "Beej Jorgensen" Hall (HTML)
 * [Build Your Own Lisp](http://www.buildyourownlisp.com) - Daniel Holden
 * [C Elements of Style](http://www.oualline.com/books.free/style/) - Steve Oualline
-* [C for Python Programmers](http://www.cburch.com/books/cpy/) - Carl Burch
+* [C for Python Programmers](https://www.cs.toronto.edu/~patitsas/cs190/c_for_python.html) - Carl Burch
```
Comment (davorpa) [inline books/free-programming-books-langs.md:432]: What is the problem here? http://www.cburch.com/books/cpy/ is still available

Comment (Spokaren) [reply books/free-programming-books-langs.md:432]: is return myself "Site not found" in dreamhost page

Comment (Spokaren) [reply books/free-programming-books-langs.md:432]: even for the domain page

File: courses/free-courses-fr.md (hunk 1 above)
Comment (Spokaren) [reply courses/free-courses-fr.md:26]: yes, that because I used the same branch

File: books/free-programming-books-langs.md (hunk 2 above)
Comment (eshellman) [reply books/free-programming-books-langs.md:432]: works fine for me too

Comment (Spokaren) [reply books/free-programming-books-langs.md:432]: I connect myself to the website with https by default, my bad..

Comment (Spokaren) [reply books/free-programming-books-langs.md:432]: I going to close this pr since this is not a issue. Thank for your review.

**Pull request closed without merging**